app/src/main/java/sleepyweasel/purplefluffernutter/EditMovieActivity.java

```diff
@@ -51,6 +51,18 @@ public void goToList(View v) {
         startActivity(i);
     }
 
+    public void onClickSaveButton(View v) {
+        addNewMovieToContentProvider();
+        goToDetail(v);
+    }
+
+    public void addNewMovieToContentProvider() {
+        TextView text = (TextView) findViewById(R.id.movie_title_value);
+        String movieTitle = text.getText().toString();
+        MovieEntry movieEntry = new MovieEntry(movieTitle);
+        MovieContentProvider.getInstance().addMovie(movieEntry);
```

Comment:
albo podepnij daggera 2 albo olej. Lepiej olej. ja to najwyzej zrobie jak sie skusze na podpiecie.